src/interpreter.c

```diff
@@ -94,9 +94,7 @@ static jl_value_t *eval_methoddef(jl_expr_t *ex, interpreter_state *s)
             jl_error("method: invalid declaration");
         }
         jl_binding_t *b = jl_get_binding_for_method_def(modu, fname);
-        _Atomic(jl_value_t*) *bp = &b->value;
-        jl_value_t *gf = jl_generic_function_def(fname, modu, bp, b);
-        return gf;
+        return jl_declare_const_gf(b);
```

Comment:
It is typically better to pass `modu` and `fname` here explicitly, rather than relying on extracting something from `b` and hoping it is relevant, as they aren't necessarily identical to the value in `b` if it was imported with a different name (or in the future, some Binding might not have a name or module associated at all)

Reply:
I'll change that in the future. It's a little awkward right now with the way that binding lookup is done in codegen.

Reply:
What was wrong with keeping the existing arguments in place here?